engine/storage/volume/src/org/apache/cloudstack/storage/volume/VolumeServiceImpl.java

```diff
@@ -1178,7 +1184,8 @@ public boolean destroyVolume(long volumeId) throws ConcurrentOperationException
         try {
             DataObject volumeOnStore = store.create(volume);
             volumeOnStore.processEvent(Event.CreateOnlyRequested);
-            snapshot.processEvent(Event.CopyingRequested);
```

Comment:
@jayapalu
Snapshot has derived entity(s) in snapshot_store_ref table. While creating volume from the snapshot request , state transition processed on this derived entity instead of Actual Entity. However the state transition happening on this derived entity is not observable in UI(Browser). Since this transition has no specific usage across the product, truncated the transition.

Earlier due to this transition when a request is in progress , if new volume create request comes and fails because there is no transition state available. Thereafter removal of process event is enabling parallel volume creation in ACS.